dig.go

```diff
@@ -258,6 +265,35 @@ func setRand(r *rand.Rand) Option {
 	})
 }
 
+// DryRun is an Option which, when set to true, disables invocation of functions supplied to
+// Provide and Invoke. Use this to build no-op containers.
+func DryRun(dry bool) Option {
+	return optionFunc(func(c *Container) {
+		if dry {
+			c.invokerFn = dryInvoker
+		} else {
+			c.invokerFn = defaultInvoker
+		}
+	})
+}
+
+// invokerFn specifies how the container calls user-supplied functions.
+type invokerFn func(fn reflect.Value, args []reflect.Value) (results []reflect.Value)
+
+func defaultInvoker(fn reflect.Value, args []reflect.Value) []reflect.Value {
+	return fn.Call(args)
+}
+
+// Generates zero values for results without calling the supplied function.
+func dryInvoker(fn reflect.Value, _ []reflect.Value) []reflect.Value {
+	results := make([]reflect.Value, fn.Type().NumOut())
+	for i := 0; i < fn.Type().NumOut(); i++ {
+		results[i] = reflect.Zero(fn.Type().Out(i))
```

Comment:
minor: extract fn.Type() into a local variable.

```suggestion
	ft := fn.Type()
	results := make([]reflect.Value, ft.NumOut())
	for i := 0; i < ft.NumOut(); i++ {
		results[i] = reflect.Zero(ft.Out(i))
```